### PR TITLE
Ports NSV13's Nuclear Reactor + Pools

### PR DESCRIPTION
**Pull request by maxion12345**
- [ ] Fix power production
- [ ] Figure out a way to make the startup faster/less boring
- [ ] Alternative alarm? Station-wide beeping kinda weird. SM radio style?
- [ ] Test gas interactions/add a turbine system for when you run it hot hot

# Document the changes in your pull request
Ports the Advanced Gas Nuc- fuck it its the nuclear reactor
I also ported pools with it because I needed a spent fuel pool

basically this reactor uses nuclear fuel to make gas
there's a moderator to control the reaction
and a coolant to keep it cool
and it can meltdown/fucking explode

still a WIP because it basically doesn't output power, not sure if I need to add turbines myself or what

# Wiki Documentation

steal NSV's reactor guide

# Changelog

Edit the changelog below to reflect the changes made by this PR, even if the changes are minor - required for every PR.
If you add a name after the ':cl', that name will be used in the changelog. Leave it empty to use your GitHub name.

:cl:  
rscadd: Added new things  
rscdel: Removed old things  
bugfix: fixed a few things  
wip: This is work in progress  
tweak: tweaked a few things  
soundadd: added a new sound thingy  
sounddel: removed an old sound thingy  
imageadd: added some icons and images  
imagedel: deleted some icons and images  
spellcheck: fixed a few typos  
experimental: This is experimental  
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
> there's a moderator to control the reaction and a coolant to keep it 

I don't wanna do that

**Comment by ShatteredParadox** [comment]
Nice, but I believe the storm drive is much more interesting, with much more.. interesting effects. If you're adding the AGCNR, might as well go for both.

**Comment by TheLinuxOS** [comment]
> > there's a moderator to control the reaction and a coolant to keep it
> 
> I don't wanna do that

Don’t worry you’re an admin now so you won’t have to

**Comment by Xoxeyos** [comment]
The issue with this is that NSV13 has a different play-style than us, while we can kick back and let the engine run, you actually have to keep an eye on this type of reactor; I'm not sure how much you've changed, but people may be irritated with the fact that they may have to baby this.

**Comment by maxion12345** [comment]
> The issue with this is that NSV13 has a different play-style than us, while we can kick back and let the engine run, you actually have to keep an eye on this type of reactor; I'm not sure how much you've changed, but people may be irritated with the fact that they may have to baby this.

I know, it also starts WAYYY too slow. You are supposed to have it get to 60%, then hold it there, but getting it to anything above 20 takes like ten minutes. I'll have to fuck around with it. Also, the waste gas it produces cant be filtered by filtered for some reason

**Comment by patpol4** [comment]
Fixed active turfs, changed all space turfs to template turfs, fixed overlapping decals, placed turfs under blast doors, fixed inconsistent decals, placed platings under windows, replaces space windows with shutter versions, moved the cans, moved the button that vents the entire room to a more reasonable place, replaced some turfs under airlocks, moves the pool stuff out into a crate nearby, some other minor fixes.

I have fixed most of the obvious issues, this should be good for a testmerge now I guess. There are still things like how the engineering department becomes really narrow due to the way you made the template have walls up top, or the venting feature which I'm sceptical about, also I have no idea how or if this works, I just fixed the things I usually fix.

**Comment by patpol4** [comment]
yaay yogstation merge conflict on a pr that doesn't edit yogstation

**Comment by patpol4** [comment]
Should be good for a testmerge (at least from the mapping side) to keep fixing the issues, we could also use some way to let players know how this thing actually works?

**Comment by maxion12345** [comment]
> Should be good for a testmerge (at least from the mapping side) to keep fixing the issues, we could also use some way to let players know how this thing actually works?

still not finished. It doesn't actually output power, and you have to wait a long time for it get to 60% when starting. And when I went AFK and accidentally went above 60% during testing, I tried to turn it off and it spiraled out of control. Adding the "Shutdown Gas" didn't really work, so I'll test again after tweaking.

tl;dr
gotta tweak
maybe later I don't really want to look at it right now

**Comment by maxion12345** [comment]
closing for now, it's been too busy to work on. This thing really isn't a good engine at all tbh. While mechanically it's something new, the interface and generating power from it leaves much to be desired. maybe I'll come back and finish it some other time.
